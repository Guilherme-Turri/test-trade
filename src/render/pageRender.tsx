import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import store from '../storage/store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

export const pageRender = (children: React.ReactNode) => {
  return render(
    <ThemeProvider theme={theme}>
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    </ThemeProvider>);
};