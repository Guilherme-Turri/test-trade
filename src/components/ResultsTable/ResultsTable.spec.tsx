import { componentRender } from "../../render/componentRender";
import { screen } from '@testing-library/react'
import { ResultsTable } from ".";

describe('<ResultsTable/>', () => {
  const resultsMock = {
    wins: '1',
    draws: '2',
    loses: '3',
    total: '6',
  }
  it('should render on screen properly', () => {
    const { container } = componentRender(<ResultsTable {...resultsMock} />)

    const table = screen.getByTestId('resultsTable')
    const wins = screen.getByText(resultsMock.wins);
    const draws = screen.getByText(resultsMock.draws);
    const loses = screen.getByText(resultsMock.loses);
    const total = screen.getByText(resultsMock.total);
    expect(table).toBeInTheDocument()
    expect(wins).toBeInTheDocument()
    expect(draws).toBeInTheDocument()
    expect(loses).toBeInTheDocument()
    expect(total).toBeInTheDocument()
    expect(container).toMatchSnapshot();
  })

})