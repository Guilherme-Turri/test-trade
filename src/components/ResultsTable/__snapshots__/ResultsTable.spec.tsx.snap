// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsTable/> should render on screen properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 25rem;
  height: 100%;
  padding: 0.938rem;
  margin: 0.938rem;
}

.c0 th {
  padding: 0.938rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: white;
}

.c0 td {
  text-align: center;
  padding: 0.938rem;
  background-color: white;
}

.c0 caption {
  font-weight: bold;
  font-size: 1.5rem;
}

<div>
  <div
    class="c0"
    data-testid="resultsTable"
  >
    <table>
      <thead>
        <tr>
          <th>
            Vit√≥rias
          </th>
          <th>
            Derrotas
          </th>
          <th>
            Empates
          </th>
          <th>
            Total
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            3
          </td>
          <td>
            2
          </td>
          <td>
            6
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
